In [2]:
def getvoltages( pl, pswing, sswing, rgswing, drd=8.0, **kwargs ):
	pl_off = 0.0
	pu_off = 0.0
	rd_off = 0.0
	od_off = 0.0
	og_off = 0.0
	sl_off = 0.0
	sh_off = 0.0
	rgl_off = 0.0
	rgh_off = 0.0
	gd_off = 0.0
	chz = 11
	# Rules to set the voltages
	pl0 = pl			 # nominal
	pl1 = pl0 + pl_off   # corrected
	#
	pu0 = pl0 + pswing	# nominal based on pl nominal
	pu1 = pu0 + pu_off   # corrected
	#
	rd0 = pu0 + drd
	rd1 = rd0 + rd_off
	#
	od0 = rd0 + 11.8
	od1 = od0 + od_off
	#
	og0 = rd0 - chz - 4
	og1 = og0 + og_off
	#
	sl0 = og0 - 2.0
	sl1 = sl0 + sl_off
	#
	sh0 = sl0 + sswing
	sh1 = sh0 + sh_off
	#
	rgh0 = rd0 - 5.5
	rgh1 = rgh0 + rgh_off
	#
	rgl0 = rgh0 - rgswing
	rgl1 = rgl0 + rgl_off
	#
	gd0 = 26
	gd1 = gd0 + gd_off
	#
	# Apply the offsets
	# pl = pl + pl_off
	# pu = pu + pu_off
	# rd = rd + rd_off
	# od = od + od_off
	# og = og + og_off
	# sl = sl + sl_off
	# sh = sh + sh_off
	# rgh = rgh + rgh_off
	# rgl = rgl + rgl_off
	# ----------
	proto = {
			"DAC": {
				"pclkHighP": pu1,
				"pclkLowP": pl1,
				"sclkHighP": sh1,
				"sclkLowP": sl1,
				"rgHighP": rgh1,
				"rgLowP": rgl1,
			},
			"Bias": {
				"rdP": rd1,
				"odP": od1,
				"ogP": og1,
				"gdP": gd1,
			}
		}
	for key in [ "DAC", "Bias" ]:
		proto[key].update(kwargs[key])
	return proto


# standard voltage based on the rule (run5)

In [8]:
getvoltages( **{
                        "pl" : -6.0,
                        "pswing" : 9.6,
                        "sswing" : 9.3,
                        "rgswing" : 10.1,
                        "drd" : 8.0,
                        "DAC": {
                        },
                        "Bias": {
                        } }
           )

{'DAC': {'pclkHighP': 3.5999999999999996,
  'pclkLowP': -6.0,
  'sclkHighP': 3.9000000000000004,
  'sclkLowP': -5.4,
  'rgHighP': 6.1,
  'rgLowP': -4.0},
 'Bias': {'rdP': 11.6, 'odP': 23.4, 'ogP': -3.4000000000000004, 'gdP': 26.0}}

# Run6a voltage set

We empirically lowered parallel swing voltage by 0.3V from this set during Run 5. So I change pclkHighP from 3.6 to 3.3 by hand

{'DAC': {'pclkHighP': 3.3,
  'pclkLowP': -6.0,
  'sclkHighP': 3.9000000000000004,
  'sclkLowP': -5.4,
  'rgHighP': 6.1,
  'rgLowP': -4.0},
 'Bias': {'rdP': 11.6, 'odP': 23.4, 'ogP': -3.4000000000000004, 'gdP': 26.0}}

# Proposed Run6b voltage set

What Pierre suggested was to follow the rule again, but change the pswing from 9.6 to 9.3

In [12]:
getvoltages( **{
                        "pl" : -6.0,
                        "pswing" : 9.3,   # <--- changed this piece
                        "sswing" : 9.3,
                        "rgswing" : 10.1,
                        "drd" : 8.0,
                        "DAC": {
                        },
                        "Bias": {
                        } }
           )

{'DAC': {'pclkHighP': 3.3000000000000007,
  'pclkLowP': -6.0,
  'sclkHighP': 3.6000000000000014,
  'sclkLowP': -5.699999999999999,
  'rgHighP': 5.800000000000001,
  'rgLowP': -4.299999999999999},
 'Bias': {'rdP': 11.3, 'odP': 23.1, 'ogP': -3.6999999999999993, 'gdP': 26.0}}